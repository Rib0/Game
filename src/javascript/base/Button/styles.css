.button {
    padding: 10px 20px;
    outline: none;
    background: none;
    color: var(--colorWhite);
    border: 3px solid var(--colorWhite);
    border-radius: 10px;
    font-size: 20px;
    font-weight: bold;
    cursor: pointer;

    &:hover {
        color: var(--colorOrange);
        border-color: var(--colorOrange);
    }
}
