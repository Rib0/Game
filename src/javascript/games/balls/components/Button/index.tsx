import React from 'react';
import styled from 'styled-components';

import Icon, { IIconProps } from '../Icon';

interface IStyledButtonProps {
    onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
    icon?: IIconProps['icon'];
    text?: string;
    disabled?: boolean;
}

const Button = ({
    icon,
    text,
    ...rest
}: IStyledButtonProps & React.HTMLAttributes<HTMLButtonElement>) => {
    return (
        <button type="button" {...rest}>
            {icon ? <Icon icon={icon} /> : text}
        </button>
    );
};

const StyledButton = styled(Button)`
    cursor: pointer;
    padding: 5px 20px;
    border: none;
    border-radius: 8px;
    background-color: ${({ disabled }) => (disabled ? 'lightgray' : '#f6d515')};
    box-shadow: 0 4px #ac5d0b;
    pointer-events: ${({ disabled }) => (disabled ? 'none' : 'all')};

    &:active {
        transform: translateY(1px);
        box-shadow: 0 5px #ac5d0b;
    }
`;

export default StyledButton;
