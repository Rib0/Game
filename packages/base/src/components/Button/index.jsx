import React from 'react';
import classNames from 'classnames/bind';

import Icon from '@/components/Icon';

import styles from './styles.css';

const cx = classNames.bind(styles);

const TYPES = {
    default: 'default',
    styled: 'styled',
};

const SIZES = {
    medium: 'medium',
    large: 'large',
}

export default (props) => {
    const { type, size, iconType, text, className, iconClassName, ...otherProps } = props;

    const buttonClassName = cx('button', iconType && 'buttonWithIcon', type, size, className);

    return (
        <button type="button" className={buttonClassName} {...otherProps}>
            {iconType ? <Icon type={iconType} className={iconClassName} /> : text}
        </button>
    )
}

// Button.propTypes = {
//     type: PropTypes.oneOf(Object.values(TYPES)),
//     size: PropTypes.oneOf(Object.values(SIZES)),
//     iconType: PropTypes.oneOf(Object.values(Icon.IconTypes)),
//     iconSize: PropTypes.oneOf(Object.values(Icon.IconSizes)),
//     text: PropTypes.string,
//     onClick: PropTypes.func,
//     className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
//     iconClassName: PropTypes.string,
// };