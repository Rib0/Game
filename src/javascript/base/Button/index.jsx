import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import Icon from '../Icon';

import styles from './styles.css';

const cx = classNames.bind(styles);

class Button extends PureComponent {
    static Types = {
        default: 'default',
        styled: 'styled',
    };

    static Sizes = {
        medium: 'medium',
        large: 'large',
    };

    render() {
        const {
            type,
            size,
            iconType,
            text,
            className,
            iconClassName,
            children,
            ...otherProps
        } = this.props;

        const buttonClassName = cx('button', iconType && 'buttonWithIcon', type, size, className);

        return (
            <button className={buttonClassName} {...otherProps}>
                {iconType ? <Icon type={iconType} className={iconClassName} /> : text}
            </button>
        );
    }
}

Button.defaultProps = {
    type: Button.Types.styled,
    size: Button.Sizes.medium,
};

Button.propTypes = {
    type: PropTypes.oneOf(Object.values(Button.Types)),
    size: PropTypes.oneOf(Object.values(Button.Sizes)),
    iconType: PropTypes.oneOf(Object.values(Icon.IconTypes)),
    iconSize: PropTypes.oneOf(Object.values(Icon.IconSizes)),
    text: PropTypes.string,
    onClick: PropTypes.func,
    className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    iconClassName: PropTypes.string,
    children: PropTypes.element,
};

export default Button;
