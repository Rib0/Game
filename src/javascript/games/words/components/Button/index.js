import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import styles from './styles.css';

const cx = classNames.bind(styles);

const Button = props => {
    const { text, onClick, className } = props;

    const buttonClassName = cx('button', className);

    return (
        <button className={buttonClassName} onClick={onClick}>
            {text}
        </button>
    );
};

Button.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Button;
