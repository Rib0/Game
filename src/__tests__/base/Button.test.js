import React from 'react';
import Button from 'base/Button';

const setUp = props => shallow(<Button {...props} />);

describe('Button test', () => {
    const props = {
        type: Button.Types.styled,
        size: Button.Sizes.large,
        iconType: 'bars',
        className: 'customClassName',
    };

    test('Button should render', () => {
        const Component = setUp(props);
        expect(Component).toMatchSnapshot();
    });
    test('Button should render icon', () => {
        const Component = setUp(props);
        expect(Component.find('Icon')).toHaveLength(1);
    });
    test('Button should render text', () => {
        const newProps = {
            ...props,
            iconType: undefined,
            text: 'some text',
        };

        const Component = setUp(newProps);
        expect(Component.text()).toBe('some text');
    });
    test('Button should have custom className', () => {
        const Component = setUp(props);
        expect(Component.hasClass('customClassName')).toBeTruthy();
    });
    test('Button should throw other props', () => {
        const newProps = {
            ...props,
            ariaHidden: true,
        };

        const Component = setUp(newProps);
        expect(Component.prop('ariaHidden')).toBeTruthy();
    });
});
