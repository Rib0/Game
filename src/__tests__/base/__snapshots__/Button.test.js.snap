// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button should render 1`] = `
<button
  className="button buttonWithIcon styled large customClassName"
  type="button"
>
  <Icon
    size="3x"
    type="bars"
  />
</button>
`;
